index.bs

```diff
@@ -16,13 +16,443 @@ WPT Display: closed
+
+These challenges can be mitigated by supporting a new pair of headers. In particular, this document introduces:
+
+* [:Sec-Fetch-Storage-Access:], a request header to convey information about whether cross-site cookies were included in the request, and possibly whether the <a permission><code>storage-access</code></a> permission has been granted.
+* [:Activate-Storage-Access:], a response header that can be used to activate an existing <a permission><code>storage-access</code></a> permission grant and "retry" the request, or to activate an existing <a permission><code>storage-access</code></a> prior to loading a [=Document=] (typically, an <{iframe}>).
+
+Infra {#infra}
+==============
+
+This specification depends on the Infra standard. [[!INFRA]]
+
+Storage-Access Request Infrastructure {#request-infrastructure}
+===============================================================
+
+In addition to the new headers themselves, this document introduces some new infrastructure to store and convey metadata in the [=user agent=], particularly on a [=request=].
+
+A [=request=] has a boolean [=struct/item=] <dfn for="request">eligible for storage-access</dfn>. This indicates whether the relevant third-party site allows the relevant "<a permission><code>storage-access</code></a>" permission grant to be used to allow access to cookies, if it exists. Unless stated otherwise it is false.
+
+NOTE: The [=user agent=] is expected to read a [=request=]'s [=request/eligible for storage-access=] boolean when attaching cookies to an outgoing [=request=]. It is not yet possible to specify this, since the cookie store has no defined interface that specs can interact with.
+
+NOTE: A [=request=] also has a [=request/client=], which is an [=environment settings object=] (a kind of [=environment=]) and therefore has a [=environment/has storage access=] boolean. That value is distinct from the value of the [=request=]'s [=request/eligible for storage-access=] boolean.
+
+A [=request=] has an associated <dfn for="request">single-hop cache mode</dfn>, whose value is null or a [=request/cache mode=]. 
+
+This document renames a [=request=]'s [=request/cache mode=] field to <dfn for="request">internal cache mode</dfn>.
+
+<div algorithm="request cache mode">
+This document redefines a [=request=]'s <dfn for="request" export>cache mode</dfn> as the cache mode returned by running the following steps, given a [=request=] |request|:
+    1. If |request|'s [=request/single-hop cache mode=] is not null, return |request|'s [=request/single-hop cache mode=].
+    1. Return |request|'s [=request/internal cache mode=].
+</div>
+
+A <dfn>storage access status</dfn> is one of "`none`", "`inactive`", or "`active`".
```

Comment:
Not a huge deal, but these days I always recommend to make these linkable, so that they are more formally enum-like. See https://wicg.github.io/attribution-reporting-api/#eligibility-unset and https://html.spec.whatwg.org/C#cross-origin-isolation-mode.